api/graphql/types/reservation_unit/types.py

```diff
@@ -206,3 +209,12 @@ def resolve_reservable_time_spans(
                 start_date, end_date
             )
         ]
+
+    def resolve_num_active_user_reservations(root: ReservationUnit, info: GQLInfo) -> int:
+        """
+        Number of active reservations made by the user to this ReservationUnit.
+        This is used to determine if the user can make a new reservation based on the max_reservations_per_user.
+        """
+        if not info.context.user.is_authenticated:
+            return 0
+        return root.reservation_set.filter(user=info.context.user).active().count()
```

Comment:
The issue with this kind of resolver is that it will break optimization since it's evaluated for each reservation unit separately. Of course, this can't be a simple annotation either, since it needs to know about the current request user. If this is only used for single items, it can be fine, since then there is only a single additional query, but if it's used for lists, then it should be optimized somehow.

Reply:
Yes, this is intended to be only be used for single reservation unit queries by the frontend.
While nothing is preventing it to be used for other queries too, I came to the same conclusion that this should be fine since its just one additional query and optimising it might be a difficult

Reply:
I'll think about adding some mechanism to the optimizer library for this, but this is ok for now.